### PR TITLE
8 add docstrings

### PR DESCRIPTION
**Pull request by Veldhoen**
Add docstrings to the existing modules. Also add TODO's where code improvements are expedient

### CODE REVIEW

File: io_util.py [hunk 1]
```diff
@@ -297,8 +321,9 @@ def fetch_input_s3_uri(handler, doc: Document) -> str:
     return ""
```
Comment (Veldhoen) [inline io_util.py:321]: I'm not quite sure what this function is doing exactly and what its arguments are supposed to be.. @jblom could you elaborate (perhaps in the form of a docstring)?

**Comment by Veldhoen** [comment]
@greenw0lf OR @Martaesplo - please let us know if you still find docstrings you can't understand
@jblom please have a look to see if anything DANE-related makes sense & respond to my question about io_util.py

**Comment by jblom** [comment]
@Veldhoen put a review in my agenda for next Monday!

**Review by greenw0lf — APPROVED** [review]
The docstrings look sufficient and clear to me. Approved!

**Pull request merged**